AppCenter/AppCenterTests/MSOneCollectorChannelDelegateTests.m

```diff
@@ -251,6 +251,39 @@ - (void)testDidEnqueueLogToOneCollectorChannelWhenLogHasTargetTokensAndLogIsNotC
                                }];
 }
 
+- (void)testDidEnqueueLogToOneCollectorChannelSynchronously {
+
+  // If
+  id<MSChannelUnitProtocol> channelUnitMock = OCMProtocolMock(@protocol(MSChannelUnitProtocol));
+  OCMStub([channelUnitMock configuration]).andReturn(self.baseUnitConfig);
+  id<MSChannelGroupProtocol> channelGroupMock = OCMProtocolMock(@protocol(MSChannelGroupProtocol));
+  id<MSChannelUnitProtocol> oneCollectorChannelUnitMock = OCMProtocolMock(@protocol(MSChannelUnitProtocol));
+  OCMStub(oneCollectorChannelUnitMock.logsDispatchQueue).andReturn(self.logsDispatchQueue);
+  OCMStub([channelGroupMock addChannelUnitWithConfiguration:OCMOCK_ANY withIngestion:OCMOCK_ANY]).andReturn(oneCollectorChannelUnitMock);
+  NSMutableSet *transmissionTargetTokens = [NSMutableSet new];
+  [transmissionTargetTokens addObject:@"fake-transmission-target-token"];
+  MSCommonSchemaLog *commonSchemaLog = [MSCommonSchemaLog new];
+  id<MSMockLogWithConversion> mockLog = OCMProtocolMock(@protocol(MSMockLogWithConversion));
+  OCMStub([mockLog toCommonSchemaLogs]).andReturn(@[ commonSchemaLog ]);
+  OCMStub(mockLog.transmissionTargetTokens).andReturn(transmissionTargetTokens);
+  dispatch_semaphore_t sem = dispatch_semaphore_create(0);
+  /*
+   * Make sure that the common schema log is enqueued synchronously by putting a task on the log queue that won't return
+   * by the time verify is called.
+   */
+  dispatch_async(oneCollectorChannelUnitMock.logsDispatchQueue, ^{
```

Comment:
Why do we need this one? The unit test still passes without this one. Either unit test is wrong or semaphore isn't necessary.

Reply:
Though the test passes without this, it doesn't test the right thing. This guarantees that the code is not running on the dispatch queue. If it did run on the dispatch queue, it could fail.

Reply:
BTW in case of failure will the test actually fail or will it hang there forever waiting on the semaphore?

Reply:
It actually failed when we tested